dask/dataframe/io/tests/test_parquet.py

```diff
@@ -616,9 +616,9 @@ def test_optimize(tmpdir, c):
 def test_roundtrip_from_pandas(tmpdir, write_engine, read_engine):
     fn = str(tmpdir.join("test.parquet"))
     df = pd.DataFrame({"x": [1, 2, 3]})
-    df.to_parquet(fn, engine=write_engine)
+    df.to_parquet(fn, index=True, engine=write_engine)
     ddf = dd.read_parquet(fn, engine=read_engine)
-    assert_eq(df, ddf)
+    assert_eq(df, ddf, check_index=False)
```

Comment:
Does this need a TODO to reenable `check_index=True` after it's fixed in pyarrow?

Reply:
While I agree with you, I don't believe there is any problem to be "fixed" in pyarrow.  Explicitly setting `index=True` in the pandas `to_parquet` call seems to behave correctly for the `pyarrow` engine.   The reason for `check_index=False` is just that there seems to be some inconsistancies between the behavior/assumptions of the pyarrow and fastparquet engines regarding index storage/detection.

Another option is to explicitly specify the index by name:

```p
    df.index.name = 'index'
    df.to_parquet(fn, index=True, engine=write_engine)
    ddf = dd.read_parquet(fn, index='index', engine=read_engine)
    assert_eq(df, ddf)
```

Perhaps that is a cleaner fix?

Reply:
Sorry, I may have misread the original test. You can probably restore https://github.com/dask/dask/pull/5018/commits/7fe83fbfd760dcfe7b6625b569429b61f74402c9

If you do keep this change, then I'd recommend copying `df` before setting the `.index.name`, since that may affect other tests.

Reply:
ahh good point